jobs/common/pre-reqs.yaml

```diff
@@ -26,20 +27,104 @@
 - builder:
     name: 'install-common-tools'
     builders:
-      - shell:
+      - shell: |-
           #!/bin/bash
-          set -ex
+          set -eux
+
+          echo "Updating packages"
+          sudo DEBIAN_FRONTEND=noninteractive apt-get update
 
           echo "Installing common tools"
           sudo DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
             make
 
+- builder:
+    name: 'install-go'
+    builders:
+    - shell:
+        !include-raw: "scripts/install-go.sh"
+
+# setup-go-environment is responsible for setting up the basic GO environment
+# needed to build and run go applications related to juju.
+#
+# This builder well set and make the GOPATH
+- builder:
+    name: 'setup-go-environment'
+    builders:
+      - install-go
+      - shell: |-
+          #!/bin/bash
+          set -eux
+
+          touch go-env.properties
+          echo "GOPATH=${WORKSPACE}/go-path" >> go-env.properties
+
+          cat go-env.properties
+      - inject:
+          properties-file: go-env.properties
+      - shell: |-
+          #!/bin/bash
+          set -ex
+          mkdir -p $GOPATH
+          rm -rf go-env.properties
```

Comment:
Can't this just be part of the previous shell build step? I guess you want to be able to run `rm go-env.properties`
We could use `properties-content` instead and not need an intermediate file. So have that as the first step and then the shell to create the dir. No properties file needed.